src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,387 @@
+          gap: get('space.2'),
+          flexGrow: '1',
+          justifyContent: 'right',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+// PageHeader.TitleArea: The main title area of the page. Visible on all viewports.
+// PageHeader.TitleArea Sub Components: PageHeader.LeadingAction, PageHeader.LeadingVisual, PageHeader.Title, PageTitle.TrailingVisual, PageHeader.TrailingAction, PageHeader.Actions
+// PageHeader.LeadingAction and PageHeader.TrailingAction are only visible on regular viewports therefore they come as visible on narrow viewports and their visibility can be managed by their exposed `visible` prop
+// ---------------------------------------------------------------------
+const TitleArea: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {display: isHidden ? 'none' : 'flex', flexDirection: 'row', alignItems: 'center', gap: '0.5rem'},
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const LeadingAction: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  hidden = hiddenOnNarrow,
+}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return <Box sx={merge<BetterSystemStyleObject>({display: isHidden ? 'none' : 'flex'}, sx)}>{children}</Box>
+}
+
+const LeadingVisual: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, hidden = false}) => {
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    <Box
+      sx={merge<BetterSystemStyleObject>(
+        {
+          display: isHidden ? 'none' : 'flex',
+        },
+        sx,
+      )}
+    >
+      {children}
+    </Box>
+  )
+}
+
+export type TitleProps = {
+  variant?: 'subtitle' | 'medium' | 'large' | ResponsiveValue<'subtitle' | 'medium' | 'large'>
+  // Check if we need responsive values for heading is so should we update as prop's type for Heading component?
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & PageHeaderProps
+
+const Title: React.FC<React.PropsWithChildren<TitleProps>> = ({
+  children,
+  sx = {},
+  variant = 'medium',
+  hidden = false,
+  as = 'h3',
+}) => {
+  const currentVariant = useResponsiveValue(variant, 'medium')
+  const isHidden = useResponsiveValue(hidden, false)
+  return (
+    // TODO: clarify if we want rem for line height or not
+    <Heading
+      as={as}
+      sx={merge<BetterSystemStyleObject>(
+        {
+          fontSize: {
+            large: '2rem',
+            medium: '1.25rem',
+            subtitle: '1.25rem',
+          }[currentVariant],
+          lineHeight: {
+            large: '48px',
+            medium: '32px',
+            subtitle: '32px',
+          }[currentVariant],
+          fontWeight: {
+            large: '400',
+            medium: '400',
+            subtitle: '400',
+          }[currentVariant],
```

Comment:
Let's use [typography design tokens](https://primer.style/primitives/typography) instead of hard-coded strings. We can pair if you're having a tough time figuring out which tokens to use.

Reply:
As per suggested [here](https://github.com/primer/react/pull/2265#discussion_r1041511281), I'll keep the hard-coded rem here but I am confused about rem values [here](https://primer.style/primitives/typography) for line-height. The output values seem to be like `calc(48/32)` (for large text) would that mean I would use `1.5rem` for large text line-height?

Reply:
@mperrotti ^^

Reply:
No, I think we'd just use `1.5` because line-heights can be unit-less. The `rem` unit refers to the *root* font size (the font size set on `<html>`), so instead of `1.5 * {font size of element}` it would be `1.5 * {font size set on the document root}`

Reply:
Ohh that makes prefect sense!! I was thinking taking the root font-size as an anchor to proportionally scale the line-height but no we should anchor the font-size of the element. Thanks for clarifying 💖